activiti-cloud-query-service/activiti-cloud-services-query/activiti-cloud-services-query-events/src/test/java/org/activiti/cloud/services/query/events/handlers/BaseBPMNActivityEventHandlerTest.java

```diff
@@ -0,0 +1,216 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class BaseBPMNActivityEventHandlerTest {
+
+    @Mock
+    private EntityManager entityManager;
+
+    @Test
+    public void findOrCreateBPMNActivityEntity_shouldNotCreateEntityForServiceTask() {
+        // given
+        TestBaseBPMNActivityEventHandler handler = new TestBaseBPMNActivityEventHandler(entityManager);
+
+        CloudBPMNActivityImpl serviceTaskActivity = createBPMNActivity(SERVICE_TASK_TYPE);
+        CloudBPMNActivityStartedEventImpl event = createEvent(serviceTaskActivity);
+
+        // when
+        Optional<BaseBPMNActivityEntity> result = handler.findOrCreateBPMNActivityEntity(event);
+
+        // then
+        assertThat(result).isEmpty();
+        verify(entityManager, never()).find(any(), any());
+    }
+
+    @Test
+    public void findOrCreateBPMNActivityEntity_shouldCreateEntityForUserTask() {
+        // given
+        TestBaseBPMNActivityEventHandler handler = new TestBaseBPMNActivityEventHandler(entityManager);
+
+        CloudBPMNActivityImpl userTaskActivity = createBPMNActivity("userTask");
+        CloudBPMNActivityStartedEventImpl event = createEvent(userTaskActivity);
+
+        when(entityManager.find(eq(BPMNActivityEntity.class), any(String.class))).thenReturn(null);
+
+        // when
+        Optional<BaseBPMNActivityEntity> result = handler.findOrCreateBPMNActivityEntity(event);
+
+        // then
+        assertThat(result).isPresent();
+        assertThat(result.get().getActivityType()).isEqualTo("userTask");
+        assertThat(result.get().getElementId()).isEqualTo("element-123");
+        assertThat(result.get().getProcessInstanceId()).isEqualTo("process-instance-123");
+    }
+
+    @Test
+    public void findOrCreateBPMNActivityEntity_shouldFindExistingEntityForUserTask() {
+        // given
+        TestBaseBPMNActivityEventHandler handler = new TestBaseBPMNActivityEventHandler(entityManager);
+
+        CloudBPMNActivityImpl userTaskActivity = createBPMNActivity("userTask");
+        CloudBPMNActivityStartedEventImpl event = createEvent(userTaskActivity);
+
+        BPMNActivityEntity existingEntity = new BPMNActivityEntity(
+            "serviceName",
+            "serviceFullName",
+            "serviceVersion",
+            "appName",
+            "appVersion"
+        );
+        existingEntity.setId("existing-id");
+        existingEntity.setActivityType("userTask");
+
+        when(entityManager.find(eq(BPMNActivityEntity.class), any(String.class))).thenReturn(existingEntity);
+
+        // when
+        Optional<BaseBPMNActivityEntity> result = handler.findOrCreateBPMNActivityEntity(event);
+
+        // then
+        assertThat(result).isPresent();
+        assertThat(result.get()).isEqualTo(existingEntity);
```

Comment:
```suggestion
        assertThat(result).contains(existingEntity);
```

When asserting on Optionals content it's more explicit to mention they `contain` the expected entity rather than check for equality by via `get()`.

Reply:
Thanks!